### PR TITLE
temporary directory for `$HOME` is required for stack to compile

### PR DESCRIPTION
**Pull request by calvinchengx**

### CODE REVIEW

**Comment by peti** [comment]
If you'll read the first line of the file you've edited, then you'll find that this patch is not going to have any lasting effects.

Instead, the proper way to apply this patch is the same approach as https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/configuration-common.nix#L855.

**Comment by calvinchengx** [comment]
Ah, I see. So, I have to declare `stack` in `haskell-modules/configuration-common` and apply my preconfigure. Got it. Will resend.

**Comment by calvinchengx** [comment]
@peti build failed with a strange out of memory error and it says "Please disregard the result of this Travis build."  How can we run this build again?

**Comment by peti** [comment]
Just forget about Travis. Those builds are almost entirely worthless and hardly ever produce meaningful results.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -702,7 +703,6 @@ self: super: {
     enableSharedExecutables = false;
     postInstall = ''            # install man pages
       mv man $out/
-      find $out/man -type f ! -name "*.[0-9]" -exec rm {} +
```
Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: Why did you remove that line?

**Comment by calvinchengx** [comment]
@peti my mistake. rebased it and included this line back in.
